corehq/apps/users/views/mobile/users.py

```diff
@@ -97,6 +99,7 @@
                          "ManageCommCareMobileWorkers-B.UseBulkUploadtocreatem"
                          "ultipleusersatonce")
 DEFAULT_USER_LIST_LIMIT = 10
+BAD_MOBILE_USERNAME_REGEX = re.compile("[^A-Za-z.+-_]")
```

Comment:
Do we not allow numbers?

Reply:
Yep. Jenny mentioned that earlier.  Now with tests! https://github.com/dimagi/commcare-hq/pull/20868/commits/1564758ac5f6b40f8e1f1cdf449b279cbfe113d7